src/mscorlib/src/System/TimeSpan.cs

```diff
@@ -405,6 +405,42 @@ public String ToString(String format, IFormatProvider formatProvider)
             return t1.Add(t2);
         }
 
+        public static TimeSpan operator *(TimeSpan timeSpan, double factor)
+        {
+            if (double.IsNaN(factor))
+            {
+                throw new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"), nameof(factor));
+            }
+
+            double ticks = timeSpan.Ticks * factor;
+            if (ticks > long.MaxValue | ticks < long.MinValue)
+            {
+                throw new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
+            }
+
+            return FromTicks((long)ticks);
```

Comment:
I think, on balance, that rounding to the nearest tick is probably the best approach, so I'll update to do that. It's only going to matter when half a tick makes a difference, but it's probably what would have the least surprises for users.